From cb90b1cc7e4353b352391f30040099b1d7fffc1f Mon Sep 17 00:00:00 2001
From: William Roberts <william.c.roberts@intel.com>
Date: Mon, 21 Aug 2017 09:56:10 -0700
Subject: [PATCH 3/4] tpm2-command: fix cast
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixes:
src/tpm2-command.c: In function ‘tpm2_command_get_auth_attrs’:
src/tpm2-command.c:107:6: error: cast to union type from type not present in union
     ((TPMA_SESSION)(UINT32)cmd->buffer [AUTH_SESSION_ATTRS_OFFSET (cmd, index)])
      ^
Signed-off-by: William Roberts <william.c.roberts@intel.com>
---
 src/tpm2-command.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tpm2-command.c b/src/tpm2-command.c
index 05e21a9..e4896ad 100644
--- a/src/tpm2-command.c
+++ b/src/tpm2-command.c
@@ -104,7 +104,7 @@
 #define AUTH_SESSION_ATTRS_END_OFFSET(cmd, index) \
     (AUTH_SESSION_ATTRS_OFFSET (cmd, index) + sizeof (UINT8))
 #define AUTH_GET_SESSION_ATTRS(cmd, index) \
-    ((TPMA_SESSION)(UINT32)cmd->buffer [AUTH_SESSION_ATTRS_OFFSET (cmd, index)])
+    ((TPMA_SESSION)(UINT8)cmd->buffer [AUTH_SESSION_ATTRS_OFFSET (cmd, index)])
 /* authorization */
 #define AUTH_AUTH_SIZE_OFFSET(cmd, index) \
     (AUTH_SESSION_ATTRS_END_OFFSET (cmd, index))
-- 
2.7.4

